crates/red_knot_python_semantic/src/semantic_index/except_handlers.rs

```diff
@@ -0,0 +1,79 @@
+use std::cell::{RefCell, RefMut};
+
+use super::use_def::FlowSnapshot;
+use super::SemanticIndexBuilder;
+
+#[derive(Debug, Default)]
+pub(super) struct TryBlockContextsStack(Vec<TryBlockContexts>);
+
+impl TryBlockContextsStack {
+    pub(super) fn push_context(&mut self) {
+        self.0.push(TryBlockContexts::default());
+    }
+
+    pub(super) fn current_try_block_context(&self) -> &TryBlockContexts {
+        self.0
+            .last()
+            .expect("There should always be at least one `TryBlockContexts` on the stack")
+    }
+
+    pub(super) fn pop_context(&mut self) {
+        self.0.pop();
+    }
```

Comment:
Nit. We should assert that the popped value is never `None` because that would indicate an unbalanced `push`/`pop`.